codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Encoder.java

```diff
@@ -406,10 +405,10 @@ private void clear() {
     /**
      * Returns the hash code for the given header field name.
      */
-    private static int hash(byte[] name) {
+    private static int hash(CharSequence name) {
         int h = 0;
-        for (int i = 0; i < name.length; i++) {
-            h = 31 * h + name[i];
+        for (int i = 0; i < name.length(); i++) {
+            h = 31 * h + name.charAt(i);
```

Comment:
can we just use `name.hashCode()`?

Also this map structure in here looks very similar to the `DefaultHeaders` implementation. I wonder if we can make some of the same improvements (power of two, using `&`, remove the negative value checks in the `hash` function, etc...). We may even consider exposing the size of the hash structure as a parameter to the constructor. For folks that don't want a Dynamic table, or if they anticipate it will be small ... we could limit the size consumed by the underlying array (this would be similar to [DefaultHeaders](https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java#L123).

@nmittler @ejona86 - WDYT?